  "description": "",
  "dependencies": {
    "express": "^4.21.2",
    "node-fetch": "^3.3.2"
  },
  "devDependencies": {
    "@types/express": "^5.0.0",
    "typescript": "^5.7.3"
  }
}
cris@Cristians-MBP sample-express % vim custom.d.ts
cris@Cristians-MBP sample-express % npx tsc
routes.ts:10:37 - error TS2552: Cannot find name 'respondItemType'. Did you mean 'responseItemType'?

10   data = (await response.json()) as respondItemType[];
                                       ~~~~~~~~~~~~~~~


Found 1 error in routes.ts:10

cris@Cristians-MBP sample-express % vim route.ts
cris@Cristians-MBP sample-express % vim routes.ts
cris@Cristians-MBP sample-express % npx tsc
cris@Cristians-MBP sample-express % node index.js
node:events:502
      throw er; // Unhandled 'error' event
      ^

Error: listen EADDRINUSE: address already in use :::3000
    at Server.setupListenHandle [as _listen2] (node:net:1937:16)
    at listenInCluster (node:net:1994:12)
    at Server.listen (node:net:2099:7)
    at Function.listen (/Users/cris/sample-express/node_modules/express/lib/application.js:635:24)
    at file:///Users/cris/sample-express/index.js:34:8
    at ModuleJob.run (node:internal/modules/esm/module_job:271:25)
    at async onImport.tracePromise.__proto__ (node:internal/modules/esm/loader:547:26)
    at async asyncRunEntryPointWithESMLoader (node:internal/modules/run_main:116:5)
Emitted 'error' event on Server instance at:
    at emitErrorNT (node:net:1973:8)
    at process.processTicksAndRejections (node:internal/process/task_queues:90:21) {
  code: 'EADDRINUSE',
  errno: -48,
  syscall: 'listen',
  address: '::',
  port: 3000
}

Node.js v22.13.1
cris@Cristians-MBP sample-express % clear

cris@Cristians-MBP sample-express % node index.js
node:events:502
      throw er; // Unhandled 'error' event
      ^

Error: listen EADDRINUSE: address already in use :::3000
    at Server.setupListenHandle [as _listen2] (node:net:1937:16)
    at listenInCluster (node:net:1994:12)
    at Server.listen (node:net:2099:7)
    at Function.listen (/Users/cris/sample-express/node_modules/express/lib/application.js:635:24)
    at file:///Users/cris/sample-express/index.js:34:8
    at ModuleJob.run (node:internal/modules/esm/module_job:271:25)
    at async onImport.tracePromise.__proto__ (node:internal/modules/esm/loader:547:26)
    at async asyncRunEntryPointWithESMLoader (node:internal/modules/run_main:116:5)
Emitted 'error' event on Server instance at:
    at emitErrorNT (node:net:1973:8)
    at process.processTicksAndRejections (node:internal/process/task_queues:90:21) {
  code: 'EADDRINUSE',
  errno: -48,
  syscall: 'listen',
  address: '::',
  port: 3000
}

Node.js v22.13.1
cris@Cristians-MBP sample-express % vim index.js
cris@Cristians-MBP sample-express % node index.js
node:events:502
      throw er; // Unhandled 'error' event
      ^

Error: listen EADDRINUSE: address already in use :::3000
    at Server.setupListenHandle [as _listen2] (node:net:1937:16)
    at listenInCluster (node:net:1994:12)
    at Server.listen (node:net:2099:7)
    at Function.listen (/Users/cris/sample-express/node_modules/express/lib/application.js:635:24)
    at file:///Users/cris/sample-express/index.js:34:8
    at ModuleJob.run (node:internal/modules/esm/module_job:271:25)
    at async onImport.tracePromise.__proto__ (node:internal/modules/esm/loader:547:26)
    at async asyncRunEntryPointWithESMLoader (node:internal/modules/run_main:116:5)
Emitted 'error' event on Server instance at:
    at emitErrorNT (node:net:1973:8)
    at process.processTicksAndRejections (node:internal/process/task_queues:90:21) {
  code: 'EADDRINUSE',
  errno: -48,
  syscall: 'listen',
  address: '::',
  port: 3000
}

Node.js v22.13.1
cris@Cristians-MBP sample-express % node index.js
Listening on 3000

